### DISCUSSION TITLE
algorithm/%E5%9B%BE%E8%AE%BA/shu-ji-chu

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#451] [Announcements]
# algorithm/%E5%9B%BE%E8%AE%BA/shu-ji-chu

定义 有n个结点，n-1条边的联通无向图 无向无环的联通图 适用于无根树和有根树 森林：每个联通分量（连通块）都是树的图。一棵树也是森林。 生成树：一个联通无向图的生成子树，同时要求是树。即在子树中有n个点，n-1条边且联通。 只适用于有根树 父亲：对于除了根节点以外的每个节点，定义为从该节点到根路径上的第二个结点。根节点没有父结点。 祖先：一个点到根...

https://demooo.top/algorithm/%E5%9B%BE%E8%AE%BA/shu-ji-chu.html

**Comment by Demomomomo** [comment]
![](http://1308117710.vod2.myqcloud.com/eddbaa7fvodcq1308117710/33ef0f17243791579305331776/0IQepsdAWKUA.gif)

**Reply by Mister-Hope** [reply]
# DO NOT USE MY REPO IN PRODUCTION!
